include/bluetooth/services/fast_pair.h

```diff
@@ -30,18 +37,82 @@ enum bt_fast_pair_adv_mode {
 
 	/** Fast Pair not discoverable advertising, hide UI indication. */
 	BT_FAST_PAIR_ADV_MODE_NOT_DISCOVERABLE_HIDE_UI_IND,
+
 	/** Number of Fast Pair advertising modes. */
 	BT_FAST_PAIR_ADV_MODE_COUNT
 };
 
+/** @brief Fast Pair advertising battery mode. Used to generate advertising packet.
+ *
+ * Battery data can be included in advertising packet only if the Fast Pair Provider is in Fast Pair
+ * not discoverable advertising mode. To prevent tracking, the Fast Pair Provider should not include
+ * battery data in the advertising packet all the time.
+ */
+enum bt_fast_pair_adv_battery_mode {
+	/** Do not advertise battery data. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_NONE,
+
+	/** Show battery data UI indication. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_SHOW_UI_IND,
+
+	/** Hide battery data UI indication. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_HIDE_UI_IND,
+
+	/** Number of Fast Pair advertising battery modes. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_COUNT
+};
+
+/** @brief Index of battery value. */
+enum bt_fast_pair_battery_value_idx {
+	/** Left bud. */
+	BT_FAST_PAIR_BATTERY_VALUE_IDX_LEFT_BUD = 0,
+
+	/** Right bud. */
+	BT_FAST_PAIR_BATTERY_VALUE_IDX_RIGHT_BUD = 1,
+
+	/** Case. */
+	BT_FAST_PAIR_BATTERY_VALUE_IDX_BUD_CASE = 2,
+
+	/** Number of battery values. */
+	BT_FAST_PAIR_BATTERY_VALUE_IDX_COUNT
+};
+
+/** @brief Fast Pair advertising config. Used to generate advertising packet. */
+struct bt_fast_pair_adv_config {
+	/** Fast Pair advertising mode. */
+	enum bt_fast_pair_adv_mode adv_mode;
+
+	/** Fast Pair advertising battery mode. */
+	enum bt_fast_pair_adv_battery_mode adv_battery_mode;
+};
```

Comment:
Can we introduce a new generic enumerator called ```bt_fast_pair_adv_ind_mode``` and make the ```bt_fast_pair_adv_mode``` consist of only two values (discoverable / not discoverable)?

```suggestion
/**  Indication mode for a Fast Pair advertising data type. */
enum bt_fast_pair_adv_ind_mode {
	/** Show UI indication for a specific Fast Pair advertising data type. */
	BT_FAST_PAIR_ADV_IND_MODE_SHOW_UI,

	/** Hide UI indication for a specific Fast Pair advertising data type. */
	BT_FAST_PAIR_ADV_IND_MODE_HIDE_UI,
};

/**  Mode for an optional Fast Pair advertising data type. */
struct bt_fast_pair_adv_optional_data_mode {
	/** If true, include this data type in the Fast Pair advertising payload. */.
	bool include;

	/** Indication mode for an option Fast Pair advertising type. */
	enum bt_fast_pair_adv_ind_mode ind_mode;
};

/** @brief Fast Pair advertising config. Used to generate advertising packet. */
struct bt_fast_pair_adv_config {
	/** Fast Pair advertising mode. */
	enum bt_fast_pair_adv_mode adv_mode;

	/** Fast Pair advertising Account Key indication mode. */
	enum bt_fast_pair_adv_ind_mode account_key_data_ind_mode;

	/** Fast Pair advertising battery data mode. */
	struct bt_fast_pair_adv_optional_data_mode battery_data_mode;
};
```

Reply:
The show/hide UI indication mode is not related to advertising account key. It is related to indicating whether a peripheral want to be paired with or not. Regarding that we have three levels of "willingnes to pair":

1. Discoverable advertising - we want to pair with everyone 
2. Not discoverable advertising with show UI indication - we want to pair only with devices that belong to known user (the one with account key)
3.  Not discoverable advertising with hide UI indication - we don't want to pair with anyone. We just advertise our presence.

That's why we have enum that have those three options. It's also worth to mention that show/hide UI indication is only relevant during not-discoverable advertisng.

Reply:
I get your point but I don't understand how the type split prevents us from having a clean division like you mentioned. The discoverable mode enables the pairing for all peers while the non-discoverable mode enables the pairing only for peers sharing the same Google account.

The show/hide indication mode controls whether notifications are displayed in the Android. It does not matter if it is in the context of battery information or pairing mode. It seems to me that it could be a generic type, used for the configuration of both advertising data segments.

Splitting the types in such a way that they have just one responsibility (e.g. Android notification control, pairing mode control) enables us to have basic building blocks from which we construct more complex features without duplicating the type information (e.g. show/hide indication for battery extension and pairing mode).

It would be great to have the type for controlling indication as a separate and independent type that can be used for building account key data payload and battery extension. I don't necessarily insist to adapt the rest of the API as I suggested with the ```bt_fast_pair_adv_config``` and ```bt_fast_pair_adv_optional_data_mode``` structures. We could consider the following items:
- Indicate optionality in another way than the ```include``` field (e.g. NULL pointer = no battery extension)
- Based on your observation that indication mode either for battery or Account Key data is only relevant for not-discoverable advertising, we can have dedicated APIs for encoding each advertising mode. e.g. ```bt_fast_pair_adv_discoverable_data_fill``` and ```bt_fast_pair_adv_not_discoverable_data_fill```
If you have any other suggestions on how to approach this API reconstruction, let me know

Reply:
I think current approach better. I see advertising mode as three state information and because of that we should represent it as enum with three possible values. I would rather not split API into two separate functions. That's what the `adv_config` parameter is for. It specifies advertising mode. Can I leave it as it is now?

Reply:
@MarekPieta

Reply:
Moved this discussion outside the scope of this PR. Please do not resolve this comment thread so that the comment URLs work correctly